File: .skills/openclaw-skills/skills/manuelhettich/ticktick/scripts/commands/abandon.ts
import { api } from "../api";

interface AbandonOptions {
  list?: string;
  json?: boolean;
}

// Check if string looks like a task ID (24-char hex)
function isTaskId(str: string): boolean {
  return /^[a-f0-9]{24}$/i.test(str);
}

export async function abandonCommand(
  taskNameOrId: string,
  options: AbandonOptions
): Promise<void> {
  try {
    let found: { task: { id: string; title: string }; projectId: string } | undefined;

    // If it looks like an ID and no --list specified, use findTaskById
    if (isTaskId(taskNameOrId) && !options.list) {
      found = await api.findTaskById(taskNameOrId);
    } else {
      // Search by title (with optional project filter)
      found = await api.findTaskByTitle(taskNameOrId, options.list);
    }

    if (!found) {
      console.error(`Task not found: ${taskNameOrId}`);
      process.exit(1);
    }

    const { task, projectId } = found;

    // Status -1 = won't do / abandoned in TickTick
    const result = await api.updateTask({
      id: task.id,
      projectId: projectId,
      status: -1,
    });

    if (options.json) {
      console.log(JSON.stringify(result, null, 2));
      return;
    }

    console.log(`✓ Abandoned: "${task.title}"`);
  } catch (error) {
    console.error(
      `Error: ${error instanceof Error ? error.message : String(error)}`
    );
    process.exit(1);
  }
}

File: .skills/openclaw-skills/skills/hfarazi/ticktickpower/ticktick/commands/abandon.py
import json
import sys

from ..api import api
from ..util import is_task_id


def abandon_command(args) -> None:
    try:
        task_name_or_id = args.task

        if is_task_id(task_name_or_id) and not args.list:
            found = api.find_task_by_id(task_name_or_id)
        else:
            found = api.find_task_by_title(task_name_or_id, args.list)

        if not found:
            print(f"Task not found: {task_name_or_id}", file=sys.stderr)
            sys.exit(1)

        task = found["task"]
        project_id = found["projectId"]

        # status -1 = won't do / abandoned in TickTick
        result = api.update_task({"id": task["id"], "projectId": project_id, "status": -1})

        if args.json:
            print(json.dumps(result, indent=2))
            return

        print(f'✓ Abandoned: "{task["title"]}"')

    except RuntimeError as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)
